packages/shorebird_cli/lib/src/commands/patch/ios_framework_patcher.dart

```diff
@@ -194,30 +165,17 @@ class IosFrameworkPatcher extends Patcher {
     final aotSnapshotFile = File(
       p.join(shorebirdEnv.getShorebirdProjectRoot()!.path, 'build', 'out.aot'),
     );
+    // TODO(eseidel): Drop support for builds before the linker.
     final useLinker = AotTools.usesLinker(shorebirdEnv.flutterRevision);
     if (useLinker) {
-      // If we're using a newer version of the linker, we need to also copy the
-      // necessary class table link information alongside the snapshots.
-      if (releaseClassTableLinkInfoFile != null &&
-          releaseClassTableLinkDebugInfoFile != null) {
-        // Copy the release's class table link info file next to the release
-        // snapshot so that it can be used to generate a patch.
-        releaseClassTableLinkInfoFile.copySync(
-          p.join(releaseArtifactFile.parent.path, 'App.ct.link'),
-        );
-        releaseClassTableLinkDebugInfoFile.copySync(
-          p.join(releaseArtifactFile.parent.path, 'App.class_table.json'),
-        );
-
-        // Copy the patch's class table link info file to the build directory
-        // so that it can be used to generate a patch.
-        File(
-          _patchClassTableLinkInfoFile,
-        ).copySync(p.join(shorebirdEnv.buildDirectory.path, 'out.ct.link'));
-        File(_patchClassTableLinkDebugInfoPath).copySync(
-          p.join(shorebirdEnv.buildDirectory.path, 'out.class_table.json'),
-        );
-      }
+      apple.copySupplementFilesToSnapshotDirs(
+        releaseSupplementDir: releaseSupplementDir,
+        releaseSnapshotDir: releaseArtifactFile.parent,
+        patchSupplementDir: Directory(
+          p.join(shorebirdEnv.buildDirectory.path, 'ios', 'shorebird'),
```

Comment:
minor cleanup would be to have `shorebirdEnv` expose an `iosSupplementDirectory` but since this is only used in one place it doesn't really matter 🤷